python/PyQt6/core/auto_generated/vector/qgsvectorlayertools.sip.in

```diff
@@ -28,7 +28,7 @@ in your application.
     QgsVectorLayerTools();
 
 
-    virtual bool addFeature( QgsVectorLayer *layer, const QgsAttributeMap &defaultValues = QgsAttributeMap(), const QgsGeometry &defaultGeometry = QgsGeometry(), QgsFeature *feature /Out/ = 0, QWidget *parentWidget = 0, bool showModal = true, bool hideParent = false ) const = 0;
+    virtual bool addFeature( QgsVectorLayer *layer, const QgsAttributeMap &defaultValues = QgsAttributeMap(), const QgsGeometry &defaultGeometry = QgsGeometry(), QgsFeature *feature /Out/ = 0, QWidget *parentWidget = 0, bool showModal = true, bool hideParent = false, QgsExpressionContextScope *scope = 0 ) const = 0;
```

Comment:
This is technically an API break -- If a plugin is implementing some custom QgsVectorLayerTools then the addFeature override will no longer apply.

We could avoid this by adding virtual addFeatureV2 which takes the new argument, and then make both addFeature/addFeatureV2 non-pure virtual, calling the other function in their base class implementations (with an explicit note in the docs that either addFeature or addFeatureV2 must be implemented).

But if we're doing this, then let's be future proof and introduce a QgsVectorLayerToolsContext class. And then the additional expression context scopes could be set on the QgsVectorLayerToolsContext object, like we do in eg QgsSymbolWidgetContext::setAdditionalExpressionContextScopes. addFeatureV2 would then have a QgsVectorLayerToolsContext argument instead of the direct QgsExpressionContextScope argument.